source/blog/2017-04-13-rpm-ostree-v2017.4-released.html.md

```diff
@@ -0,0 +1,259 @@
+---
+title: rpm-ostree v2017.4 released
+author: jlebon
+date: 2017-04-13 18:00:00 UTC
+tags: rpm-ostree, rpm, ostree, atomic, atomic host
+comments: true
+published: true
+---
+
+[rpm-ostree](https://rpm-ostree.readthedocs.org/) is the
+hybrid image/package system that provides transactional
+upgrades on Atomic Host. Here are some highlights from
+version v2017.4, as well as from earlier versions this year.
+
+READMORE
+
+### Deployment operations are now cumulative
+
+Previously, if you wanted to both install a package and
+upgrade, you would have to do:
+
+```
+# rpm-ostree upgrade
+# reboot
+# rpm-ostree install foo
+# reboot
+```
+
+Not very efficient. This was changed in v2017.2 so that
+operations became cumulative, thus only requiring one
+reboot:
+
+```
+# rpm-ostree upgrade
+# rpm-ostree install foo
+# reboot
+```
+
+This is much better, but it's still causing rpm-ostree to
+create an intermediate deployment that soon after gets
+overwritten. In v2017.4, we can finally do these operations
+in a single step:
+
+```
+# rpm-ostree upgrade --install foo
+# reboot
+```
+
+Of course, the same `--install` and `--uninstall` switches
+are available on the `deploy` and `rebase` commands.
+
+There is work underway to further improve on this and
+directly update the booted deployment, so that rebooting is
+no longer needed.
+
+### RPMs can now be directly installed
+
+Previously, if you wanted to overlay a specific RPM file,
+you'd have to shove it into a repo and place that repo in
+`/etc/yum.repos.d`. With v2017.4, rpm-ostree is now able to
+install it directly:
+
+```
+# rpm-ostree install ltrace.rpm
+# rpm-ostree status
+State: idle
+Deployments:
+  fedora-atomic:fedora-atomic/25/x86_64/docker-host
+             Version: 25.89 (2017-03-26 21:04:50)
+          BaseCommit: 6a71adb06bc296c19839e951c38dc0b71ee5d7a82262fef9612f256f0c2a70da
+              OSName: fedora-atomic
+       LocalPackages: ltrace-0.7.91-20.fc25.x86_64
+
+* fedora-atomic:fedora-atomic/25/x86_64/docker-host
+             Version: 25.89 (2017-03-26 21:04:50)
+          BaseCommit: 6a71adb06bc296c19839e951c38dc0b71ee5d7a82262fef9612f256f0c2a70da
+              OSName: fedora-atomic
+```
+
+As you can see above, the package shows up in a new
+`LocalPackages`.  This is different from `LayeredPackages`:
+packages in the latter are updated as new deployments are
+created, whereas those in the former are *never* updated. So
+in the example above, `ltrace` will remain at that version.
+
+You can always "unstick" a local package and fetch it back
+from the repos:
+
+```
+# rpm-ostree uninstall ltrace-0.7.91-20.fc25.x86_64 --install ltrace
```

Comment:
any thoughts on supporting `uninstall` for a package from the base layer? does that work now?